### ISSUE TITLE
preferences.txt changed back to default

### ISSUE THREAD

**Issue by ppeters** [gohai/processing-uploadtopi#1]
After installing the uploadtopi tool using the processing contribution manager I tried to get the 'uploadtopi' tool operational. After altering the configuration values in preferences.txt I noticed that after a fresh `start` of processing, all my handmade changes are lost.

I manually changed preferences.txt to contain:

```
gohai.uploadtopi.autostart=false
gohai.uploadtopi.hostname=192.168.1.199
gohai.uploadtopi.logging=false
gohai.uploadtopi.password=raspberry
gohai.uploadtopi.persistent=false
gohai.uploadtopi.username=pi
```

after a processing restart it showed:

```
gohai.uploadtopi.autostart=true
gohai.uploadtopi.hostname=raspberrypi.local
gohai.uploadtopi.logging=true
gohai.uploadtopi.password=raspberry
gohai.uploadtopi.persistent=true
gohai.uploadtopi.username=pi
```

I presume that is not what is intended?

**Comment by gohai** [comment]
@ppeters I presume you made the change while Processing wasn't (completely) closed? I saw this myself before: Processing writes all it's settings to file when you quit it, regardless if they got modified or not, so this would overwrite any manual modifications that were done to the file in the mean time.

In the future, the "Upload to Pi" tool will hopefully get some UI elements to modifying the settings, which renders this issue irrelevant.

**Comment by ppeters** [comment]
Gottfried,

That is probably it. Should have thought of that myself :-). I'll try it today. Thanks!

Kind regards,

## Peter

Peter Peters, LG 1.76,
Sent from phone

-----Original Message-----

**Comment by ppeters** [comment]
Gottfried,

Your solution was correct! First closing Processing and then changing preferences.txt did the trick.
That did, however, also lead to the next problem:

When I try to upload to pi, I get an exception (shown below) related to the ssh connection that is made I think.
I tried to see whether I could find a solution, but failed to do so. I created an ‘/home/pi/.ssh/unknown_hosts’
file (empty) on the Pi, but that did not help.

## Here’s the message I get in processing:

Compiling resources
Linking
Successfully created D:\Users\ppeters\Desktop\ShapeVertices\application.windows64\ShapeVertices.exe
java.io.IOException: Could not load known_hosts
               at net.schmizz.sshj.SSHClient.loadKnownHosts(SSHClient.java:621)
               at gohai.uploadtopi.UploadToPiTool.connect(UploadToPiTool.java:250)
               at gohai.uploadtopi.UploadToPiTool$1.run(UploadToPiTool.java:144)
               at java.lang.Thread.run(Thread.java:745)

## java.io.IOException: Could not load known_hosts

Kind regards,

## Peter

Peter Peters, p.j.f.peters@tue.nlmailto:p.j.f.peters@tue.nl, http://www.tue.nl/staff/p.j.f.peters, +31402473497
TU/ehttp://www.tue.nl/ Industrial Designhttp://www.tue.nl/universiteit/kolom-2/faculteiten/industrial-design/, LG 1.76, P.O. Box 513, 5600 MB Eindhoven, The Netherlands

This message was sent causing no harm to trees. However, a large number
of electrons were terribly inconvenienced
[http://www.tue.nl/fileadmin/content/faculteiten/id/Research_DI/images/TUE-DI_logo.png]

**Comment by gohai** [comment]
@ppeters Thanks for the report. This is hopefully fixed in the release I just uploaded (v1.0.5).
